### DISCUSSION TITLE
Trying to store base64 format string on blockchain

### DISCUSSION THREAD

**Discussion by qualwebs** [bitshares/bitshares-core#2810] [Q&A] [answered]
I had an idea, I was looking to store images in base64 format on blockchain, using the `account_store_map` function. 
However, the base64 is too big so I increased the `maximum_transaction_size` property of my genesis.json file. 

I wanted to implement this, but I also wanted to know that if increasing the maximum transaction size would ultimately lead to increase in the block size itself, which in turn would affect performance or might have any other negative impacts ?

**Comment by abitmore** [comment, marked as answer]
> if increasing the maximum transaction size would ultimately lead to increase in the block size itself,

Yes, likely.

> which in turn would affect performance or might have any other negative impacts ?

Likely. With the current implementation with `account_store_map`, nodes load all data into memory (RAM), IMHO it's not suitable for images. If you want to deal with images, I would recommend that you modify the plugin to not load all data into memory but only fetch images from disk when a client requests.
